std/asys/FileSystem.hx

```diff
@@ -0,0 +1,220 @@
+package asys;
+
+import haxe.Error;
+import haxe.io.Bytes;
+import haxe.io.FilePath;
+import asys.io.*;
+
+typedef FileReadStreamCreationOptions = {
+	?flags:FileOpenFlags,
+	?mode:FilePermissions
+} &
+	asys.io.FileReadStream.FileReadStreamOptions;
+
+/**
+	This class provides methods for synchronous operations on files and
+	directories. For asynchronous operations, see `asys.async.FileSystem`.
+
+	Passing `null` as a path to any of the functions in this class will result
+	in unspecified behaviour.
+**/
+extern class FileSystem {
```

Comment:
Missing `absolutePath` method. E.g.: https://api.haxe.org/sys/FileSystem.html#absolutePath

Reply:
Right, I forgot about it. Such a method should exist on the `haxe.io.FilePath` type, because it does not involve actually interacting with the OS.

Reply:
I think there is no way to get the absolutePath without asking the OS where you are right now.

Reply:
That is also true. There should still be a method that normalises a filepath, even without sys access – so it can resolve e.g. `x/../y/../z/a/b/..///..` to `z`. If the path happens to be absolute to begin with, it should remain absolute.

Reply:
What about: https://api.haxe.org/haxe/io/Path.html#normalize

Reply:
@lublak Indeed, that is exactly what I meant ^^ I forgot it already exists in the APIs.